### PR TITLE
Remove local web server & Carthage

### PR DESCRIPTION
**Pull request by eneko**
- [x] I've read, understood, and done my best to follow the [*CONTRIBUTING guidelines*](https://github.com/mindbody/conduit/blob/master/CONTRIBUTING.md).

This pull request includes (pick all that apply):

- [ ] Bugfixes
- [ ] New features
- [x] Breaking changes
- [x] Documentation updates
- [ ] Unit tests
- [ ] Other

### Summary
Conduit CI has been broken for a long time (since March 2020): 
- Last successful cron build was [10 months ago](https://travis-ci.org/github/mindbody/Conduit/builds/667694645) 
- Last successful PR build was also [10 months ago](https://travis-ci.org/github/mindbody/Conduit/builds/667774882)

This is mainly because the local web-server used for unit and integration tests was outdated. As new versions of Ruby and Python have been released, and our systems and CI servers upgraded, the local server stopped running. Quite some good tech-debt we had queued.

### Implementation
- Remove local web servers
  - httpbin has been replaced with calls to https://httpbin.org (as we originally had). Video assets have been reduced in size for faster round-trips during testing.
  - oauth-test-server has been removed and not replaced (a handful of tests have been disabled for now)
- Project and workspace files have been removed to simplify the codebase and maintenance efforts.
- Carthage support has also been removed, might be added later as XCFramework if there is public demand.
- Travis CI has been removed
- GitHub Actions have been set up for CI

### Test Plan
- Run all tests with `swift test` or from Xcode

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: CI
+
+on:
+  push:
+    branches: [ master, main, release/* ]
+  pull_request:
+    branches: [ master, main, release/* ]
```
Comment (anthony-lipscomb-dev) [inline .github/workflows/ci.yml:7]: `master` no longer exists

Comment (eneko) [reply .github/workflows/ci.yml:7]: Good call, will update

**Review by tiltem — COMMENTED** [review]
Is there anyway to not disable those tests?

**Comment by eneko** [comment]
@tiltem this subset of tests have been broken for a long time (since CI stopped running).  I've been looking for a replacement for `oauth-test-server`, but unfortunately haven't found a good solution yet. My intention is to have those tests running later on.

**Review by anthony-lipscomb-dev — APPROVED** [review]

**Pull request merged**